### PR TITLE
Add UTC functions and locale strings for SQL compiler

### PR DESCRIPTION
**Pull request by ian-legler**
Add UTCNow, UTCToday, IsUTCToday function implementations and resources
Add updated error strings for formula columns scenarios
Update unit test infra to be able to tag engine-independent compilation errors with "Compile Error"

### CODE REVIEW

**Review by MikeStall — APPROVED** [review]
:shipit:

**Comment by lesaltzm** [comment]
Nit: update the PR title to something more meaningful

**Review by lesaltzm — APPROVED** [review]
:ship:

File: src/libraries/Microsoft.PowerFx.Core/Localization/Strings.cs [hunk 1]
```diff
@@ -160,7 +160,9 @@ internal static class TexlStrings
         public static StringGetter RandBetweenArg2 = (b) => StringResources.Get("RandBetweenArg2", b);
 
         public static StringGetter AboutNow = (b) => StringResources.Get("AboutNow", b);
+        public static StringGetter AboutUTCNow = (b) => StringResources.Get("AboutNow", b);
```
Comment (bryancroteau-MSFT) [inline src/libraries/Microsoft.PowerFx.Core/Localization/Strings.cs:163]: Is this supposed to use the same resource string as the normal AboutNow function?

File: src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/IsUTCToday.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+{
+    // IsUTCToday(date: d): b
+    internal sealed class IsUTCTodayFunction : BuiltinFunction
+    {
+        // Multiple invocations may result in different return values.
+        public override bool IsStateless => false;
+        public override bool IsSelfContained => true;
+        public override bool SupportsParamCoercion => true;
+        public IsUTCTodayFunction()
+            : base("IsUTCToday", TexlStrings.AboutIsUTCToday, FunctionCategories.Information, DType.Boolean, 0, 1, 1, DType.DateTime)
```
Comment (bryancroteau-MSFT) [inline src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/IsUTCToday.cs:24]: We should not be using the same DType for both DateTime local and DateTime utc

**Review by bryancroteau-MSFT — CHANGES_REQUESTED** [review]

File: src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/IsUTCToday.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            Contracts.AssertValue(args);
+            Contracts.AssertAllValues(args);
+            Contracts.AssertValue(argTypes);
+            Contracts.Assert(args.Length == argTypes.Length);
+            Contracts.AssertValue(errors);
+            Contracts.Assert(MinArity <= args.Length && args.Length <= MaxArity);
+            bool fValid = base.CheckInvocation(args, argTypes, errors, out returnType, out nodeToCoercedTypeMap);
+            DType type0 = argTypes[0];
+            // Arg0 should not be a Time
+            if (type0.Kind == DKind.Time)
```
Comment (bryancroteau-MSFT) [inline src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/IsUTCToday.cs:42]: Should we check for DKind.Time before or after we call the base CheckInvocation (which may in fact produce a coercion and modify the nodeToCoercedTypeMap)?

**Review by bryancroteau-MSFT — CHANGES_REQUESTED** [review]

File: src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/IsUTCToday.cs (hunk 3 above)
Comment (ian-legler) [reply src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/IsUTCToday.cs:42]: This is the logic used today for IsToday

**Review by bryancroteau-MSFT — APPROVED** [review]

**Pull request merged**